control-plane/store/src/etcd.rs

```diff
@@ -111,16 +179,15 @@ fn watch(
                 Ok(msg) => {
                     match msg {
                         Some(resp) => resp,
-                        // If there is no message, continue to wait for the
-                        // next one.
-                        None => continue,
+                        // stream cancelled
+                        None => {
+                            return;
+                        }
                     }
                 }
                 Err(e) => {
-                    // If we failed to get a message, continue to wait for the
-                    // next one.
                     tracing::error!("Failed to get message with error {}", e);
-                    continue;
+                    return;
```

Comment:
Should we just return here or should we have a number of attempts before giving up?

Reply:
on the etcd-client examples they bail out immediately in case of an error. And from my testing once you get an error there's no recovery.

Reply:
Ok, that sounds fine then.